### PR TITLE
docs(developer): add developer docs and refactor README

### PR DESCRIPTION
**Pull request by chalin**
A first edition of `DEVELOPER.md` inspired from the angular.dart version, the angular.js documentation, and expanded to include instructions given in this repos’ `README.md` (which has been cleaned up and padded up).

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+## Prerequisite Software
+
+Before you can build and test Angular, you must install and configure the
+following products on your development machine:
+
+* [Dart](https://www.dartlang.org/) (version `>=1.9.0-dev.8.0`), specifically the Dart-SDK and
+  Dartium (a version of [Chromium](http://www.chromium.org) with native support for Dart through
+  the Dart VM). One of the **simplest** ways to get both is to install the **Dart Editor bundle**,
+  which includes the editor, SDK and Dartium. See the [Dart tools download page for
+  instructions](https://www.dartlang.org/tools/download.html).
```
Comment (vicb) [inline DEVELOPER.md]: may be link http://gsdview.appspot.com/dart-archive/channels/dev/release/latest/ to download dev releases

Comment (mhevery) [inline DEVELOPER.md]: Also make it clear that it is optional

**Comment by mhevery** [comment]
Love it! Thanks for helping others to get up to speed!

File: DEVELOPER.md (hunk 1 above)
Comment (chalin) [inline DEVELOPER.md]: Actually, the download page that is being linked on line 27 is inclusive. I.e., it also provides a link to dev channel versions (via the archive page):
> Want another Dart version? Use the [download archive](https://www.dartlang.org/tools/download-archive). It lets you choose specific versions of the Dart downloads. It also has the latest **dev channel** builds, which let you try new features a few weeks earlier than the stable channel.

And the interface provided by the archive page is much nicer than gsdview.appspot -- see below. **I would avoid adding the extra link here. Agreed?**

-----

image

Comment (vicb) [inline DEVELOPER.md]: Cool! I was not aware about this. I always have troubles finding the dev versions which is why I have a bookmark on gsdview but your link is much better!

**Comment by chalin** [comment]
Ok then, anything else to be done or can this be merged? (Build is failing but I can't imagine it is due to my commit.)

**Comment by chalin** [comment]
Since the question came up again (in #982), I added an explicit mention of the stable and dev streams being available from the download archive.

**Pull request closed without merging**